plugins/modules/perlmod_install_info.py

```diff
@@ -0,0 +1,384 @@
+          manager. Defaults to C(true) if C(apt-file) executable is available.
+        type: str
+        choices:
+        - 'auto'
+        - 'true'
+        - 'false'
+        default: 'auto'
+    try_cpanm:
+        description: Specify whether to check for modules using cpanm. Defaults
+          to C(true) if C(cpanm) executable is available.
+        type: str
+        choices:
+        - 'auto'
+        - 'true'
+        - 'false'
+        default: 'auto'
+    update:
+        description: Specify whether to update package manager databases before
+          searching.
+        type: bool
+        default: false
+
+author:
+    - Jonathan Kamens (@jikamens)
+'''
+
+EXAMPLES = r'''
+# Search and fail if the package can't be found
+- name: Search for Net::DNS if it isn't already installed
+  perlmod_install_info:
+    name: Net::DNS
+  register: perlmod_info
+  failed_when: perlmod_info.missing is defined
+
+- name: Search for two modules, even if they're already installed
+  perlmod_install_info:
+    name:
+    - URI
+    - WWW::Mechanize
+  try_installed: false
+  register: perlmod_info
+
+- name: install dnf packages identified by perlmod_install_info
+  dnf:
+    name: "{{perlmod_info.dnf}}"
+  when: perlmod_info.dnf is defined
+
+- name: install yum packages identified by perlmod_install_info
+  yum:
+    name: "{{perlmod_info.yum}}"
+  when: perlmod_info.yum is defined
+
+- name: install yum packages identified by perlmod_install_info
+  apt:
+    name: "{{perlmod_info.apt}}"
+  when: perlmod_info.apt is defined
+
+- name: install cpanm packages identified by perlmod_install_info
+  cpanm:
+    name: "{{item}}"
+  with_items: "{{perlmod_info.cpanm}}"
+  when: perlmod_info.cpanm is defined
+'''
+
+RETURN = r'''
+installed:
+    description: List of modules that are already installed
+    type: list
+    elements: str
+    returned: when C(try_installed) is true and installed modules were found
+    sample: ['Net::DNS']
+dnf:
+    description: List of dnf requirements that should be installed to provide
+      at least some of the required Perl modules
+    type: list
+    elements: str
+    returned: when C(try_dnf) is true and requested modules were found in dnf
+    sample: ['perl(Net::DNS)']
+yum:
+    description: List of yum requirements that should be installed to provide
+      at least some of the required Perl modules
+    type: list
+    elements: str
+    returned: when C(try_yum) is true and requested modules were found in yum
+    sample: ['perl(Net::DNS)']
+apt:
+    description: List of apt packages that should be installed to provide at
+      least some of the required Perl modules
+    type: list
+    elements: str
+    returned: when C(try_apt) is true and requested modules were found in apt
+    sample: ['libnet-dns-perl']
+cpanm:
+    description: List of modules that should be installed via CPAN
+    type: list
+    elements: str
+    returned: when C(try_cpanm) is true and requested modules were found in
```

Comment:
Use `I()` for option names and `C()` for values:
```suggestion
    returned: when I(try_cpanm) is C(true) and requested modules were found in CPAN
```
Also note that isn't really true, since `try_cpanm=auto` can also cause this to be non-empty.

It might be better to change this to `returned: requested modules were found in CPAN` and explain this is in more detail in a second paragraph in `description`.